### ISSUE TITLE
Bot says user ID instead of user name when there's a mention.

### ISSUE THREAD

**Issue by TheCrafters001** [moonstar-x/discord-tts-bot#51]
#### :confused: Question

Hi. I noticed that whenever someone pings another user, it plays their User ID instead of just saying their name? Is there any way to prevent this?

#### :question: Additional Information

> No other additional information.

**Comment by moonstar-x** [comment]
As of right now this is the normal behavior. Mentions are received as the user ID which is why the bot is saying the user ID instead of their name.

I don't know if it would be worth to parse the mentions so the bot says their name instead of their ID since you can essentially just write their name instead of mentioning them.  Is there a reason why you mention the user rather than just writing their name?

**Comment by TheCrafters001** [comment]
> 
> 
> As of right now this is the normal behavior. Mentions are received as the user ID which is why the bot is saying the user ID instead of their name.
> 
> I don't know if it would be worth to parse the mentions so the bot says their name instead of their ID since you can essentially just write their name instead of mentioning them. Is there a reason why you mention the user rather than just writing their name?

I self-host the bot in a few servers, and sometimes some people tend to ping people on purpose, with the intent of saying the whole user id instead. It isn't a major issue, but for those who don't want to talk, it can get very annoying very fast.

**Comment by moonstar-x** [comment]
I see, so you would expect the bot to say their name instead of their ID, right?

**Comment by TheCrafters001** [comment]
> 
> 
> I see, so you would expect the bot to say their name instead of their ID, right?

Correct.

**Comment by moonstar-x** [comment]
Ok, I'm gonna put this as a feature request and try to get it done. I'll see if I can do it this week cause I've been quite busy :(

**Comment by TheCrafters001** [comment]
> 
> 
> Ok, I'm gonna put this as a feature request and try to get it done. I'll see if I can do it this week cause I've been quite busy :(

Alright! Take your time. I am in no big rush to get it fixed. :)

**Comment by L0SER8228** [comment]
If you plan to do this, then add some more features like this. Like emoji name instead of emoji id, channel name instead of channel id...

**Comment by L0SER8228** [comment]
A simple solution for this issue which is not the best but good. There is [mesage.cleanContent](https://discord.js.org/#/docs/main/v12/class/Message?scrollTo=cleanContent)

we may use this on say command
 ```js
const content = mesage.cleanContent.slice(`${this.client.prefix}say`.length);
ttsPlayer.say(content, GoogleProvider.NAME);
```

**Comment by moonstar-x** [comment]
Hi, thanks again for the suggestion.

I still haven't found enough time to work on this, I'm waiting for my semester to end in about 3 weeks so I can work on this.

In any case, as you may have seen from issue #54, all commands will be updated into slash commands and that means no message event handler.

**Comment by moonstar-x** [comment]
Fixed in #58.
